File: Longest-Palindrome.cpp
class Solution 
{
public:
    int longestPalindrome(string s) 
    {
        int count[128] = {0}; 
        for (int i = 0; i < s.size(); i++) 
        {
            count[s[i]]++;
        }
        
        int length = 0;
        bool oddFound = false;
        for (int i = 0; i < 128; i++) 
        {
            if (count[i] % 2 == 0) 
            {
                length += count[i];
            } 
            else 
            {
                length += count[i] - 1;
                oddFound = true;
            }
        }
        return oddFound ? length + 1 : length;
    }
};

File: Longest-Palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        \\\
        :type s: str
        :rtype: int
        \\\
        count = [0] * 128 
        for c in s:
            count[ord(c)] += 1
        
        length = 0
        oddFound = False
        for c in count:
            if c % 2 == 0:
                length += c
            else:
                length += c - 1
                oddFound = True
        
        if oddFound:
            length += 1
        
        return length
